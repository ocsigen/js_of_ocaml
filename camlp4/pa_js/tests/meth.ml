(* Test methods. *)

fun (obj : int) -> obj##m();;

fun (obj : < > Js.t) -> obj##m();;

fun (obj : < m : float Js.prop > Js.t) -> obj##m();;

fun (obj : < m : float Js.meth > Js.t) -> obj##m() + 1;;

fun (obj : < m : int -> int Js.meth > Js.t) -> obj##m() + 1;;

fun (obj : < m : int Js.meth > Js.t) -> obj##m(1);;

fun (obj : < m : int -> float Js.meth > Js.t) -> obj##m(1) + 1;;
