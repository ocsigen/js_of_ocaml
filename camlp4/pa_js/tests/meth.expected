

#     Characters 40-43:
  fun (obj : int) -> obj##m();;
                     ^^^
Error: This expression has type int but an expression was expected of type
         < .. > Js.t
#   Characters 25-28:
  fun (obj : < > Js.t) -> obj##m();;
                          ^^^
Error: This expression has type <  >
       It has no method m
#   Characters 43-49:
  fun (obj : < m : float Js.prop > Js.t) -> obj##m();;
                                            ^^^^^^
Error: This expression has type
         float Js.prop = < get : float; set : float -> unit > Js.gen_prop
       but an expression was expected of type 'jsoo_22f22ba7 Js.meth
#   Characters 43-51:
  fun (obj : < m : float Js.meth > Js.t) -> obj##m() + 1;;
                                            ^^^^^^^^
Error: This expression has type float but an expression was expected of type
         int
#   Characters 48-54:
  fun (obj : < m : int -> int Js.meth > Js.t) -> obj##m() + 1;;
                                                 ^^^^^^
Error: This expression has type int -> int Js.meth
       but an expression was expected of type 'jsoo_32f94eb9 Js.meth
#   Characters 41-47:
  fun (obj : < m : int Js.meth > Js.t) -> obj##m(1);;
                                          ^^^^^^
Error: This expression has type int Js.meth
       but an expression was expected of type
         'jsoo_776737d4 -> 'jsoo_593685be Js.meth
#   Characters 50-59:
  fun (obj : < m : int -> float Js.meth > Js.t) -> obj##m(1) + 1;;
                                                   ^^^^^^^^^
Error: This expression has type float but an expression was expected of type
         int
# 
